rust/src/runner/context.rs

```diff
@@ -228,3 +239,38 @@ impl TestContext {
         thread::sleep(self.naptime)
     }
 }
+
+// We implement Drop to circumvent the errors which arise from unlinking a directory for which
+// search or write permission is denied, or a flag denying delete for a file.
+impl Drop for TestContext {
+    fn drop(&mut self) {
+        let iter = walkdir::WalkDir::new(self.base_path()).into_iter();
+        for entry in iter {
+            let entry = match entry {
+                Ok(e) => e,
+                _ => continue,
+            };
+
+            let file_stat = match stat(entry.path()) {
+                Ok(s) => s,
+                _ => continue,
+            };
+
+            if entry.file_type().is_dir() && (file_stat.st_mode & Mode::S_IRWXU.bits()) == 0 {
```

Comment:
You could save a syscall by doing the file_type() check before calling stat().

Reply:
It also uses the `stat()` result to check if a flag is set to avoid calling `chflags` for nothing.

Reply:
Well yeah.  You would still need to call stat if the direntry isn't a directory.   I'm just suggesting you check that it's a directory first.  In that case you wouldn't need to stat it.